src/server/game/Misc/BanMgr.cpp

```diff
@@ -69,16 +69,30 @@ BanReturn BanMgr::BanAccount(std::string const& AccountName, std::string const&
     trans->Append(stmt);
 
     if (WorldSession* session = sWorld->FindSession(AccountID))
+    {
         if (session->GetPlayerName() != Author)
-            session->KickPlayer("Ban Account at condition 'FindSession(account)->GetPlayerName() != author'");
+        {
+            if (delay)
+                session->GetPlayer()->KickPlayer(Reason, delay_time);
+            else
+                session->KickPlayer("Ban Account at condition 'FindSession(account)->GetPlayerName() != author'");
+        }
+    }
 
     if (WorldSession* session = sWorld->FindOfflineSession(AccountID))
+    {
         if (session->GetPlayerName() != Author)
-            session->KickPlayer("Ban Account at condition 'FindOfflineSession(account)->GetPlayerName() != author'");
+        {
+            if (delay)
+                session->GetPlayer()->KickPlayer(Reason, delay_time);
+            else
+                session->KickPlayer("Ban Account at condition 'FindOfflineSession(account)->GetPlayerName() != author'");
+        }
+    }
 
     LoginDatabase.CommitTransaction(trans);
 
-    if (sWorld->getBoolConfig(CONFIG_SHOW_BAN_IN_WORLD))
+    if (sWorld->getBoolConfig(CONFIG_SHOW_BAN_IN_WORLD) && !delay)
```

Comment:
Why are excluding delayed banns? Add another config for it instead

Reply:
Because we shouldn't show ban reason in chat for delayed ban, because kick is comming after certain time.

Reply:
I'd still say add a config for it

Reply:
Why do you need config for that ? You have to skip this message if you are doing bandelay, cause this message is handled in bandelay function.

Reply:
If you keep this option with config, banned player will see reason in chat before kick out of game.

Reply:
No, keep the config default off. Perhaps someone would like the message to show.